components/org.apache.stratos.python.cartridge.agent/src/test/java/org/apache/stratos/python.cartridge.agent/test/PythonCartridgeAgentTest.java

```diff
@@ -104,97 +170,118 @@ public static void tearDown() {
-                        //throw new RuntimeException(line);
+    /**
+     * Creates an {@link org.apache.stratos.messaging.event.instance.notifier.ArtifactUpdatedEvent} object with a public
+     * repository URL
+     * @return
+     */
+    private static ArtifactUpdatedEvent createTestArtifactUpdatedEvent() {
+        ArtifactUpdatedEvent publicRepoEvent = new ArtifactUpdatedEvent();
+        publicRepoEvent.setClusterId(CLUSTER_ID);
+        publicRepoEvent.setTenantId(TENANT_ID);
+        publicRepoEvent.setRepoURL("https://bitbucket.org/testapache2211/opentestrepo1.git");
+        return publicRepoEvent;
+    }
+
+    @Test(timeout = TIMEOUT)
+    public void testPythonCartridgeAgent() {
+        Thread communicatorThread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                List<String> outputLines = new ArrayList<String>();
+                while (!outputStream.isClosed()) {
+                    List<String> newLines = getNewLines(outputLines, outputStream.toString());
+                    if (newLines.size() > 0) {
+                        for (String line : newLines) {
+                            if (line.contains("Subscribed to 'topology/#'")) {
+                                sleep(1000);
+                                // Send complete topology event
+                                log.info("Publishing complete topology event...");
+                                Topology topology = createTestTopology();
+                                CompleteTopologyEvent completeTopologyEvent = new CompleteTopologyEvent(topology);
+                                publishEvent(completeTopologyEvent);
+                                log.info("Complete topology event published");
+
+                                sleep(3000);
+                                // Publish member initialized event
+                                log.info("Publishing member initialized event...");
+                                MemberInitializedEvent memberInitializedEvent = new MemberInitializedEvent(
+                                        SERVICE_NAME, CLUSTER_ID, CLUSTER_INSTANCE_ID, MEMBER_ID, NETWORK_PARTITION_ID, PARTITION_ID
+                                );
+                                publishEvent(memberInitializedEvent);
+                                log.info("Member initialized event published");
+
+                                // Simulate server socket
+                                startServerSocket(9080);
+                            }
+                            if (line.contains("Artifact repository found")) {
+                                // Send artifact updated event
+                                publishEvent(artifactUpdatedEvent);
+                            }
+
+                            if (line.contains("Exception in thread") || line.contains("ERROR")) {
+                                //throw new RuntimeException(line);
```

Comment:
Hi @prabathabey , yes at the moment this does not serve any purpose. However the plan is to check the class of the error thrown and fail the test accordingly. This should be commented out.